.github/workflows/run-tests.yml

```diff
@@ -52,7 +52,7 @@ jobs:
             # LD_PRELOAD: /lib/x86_64-linux-gnu/libSegFault.so
             experimental: false
 
-          - distro-version: '36'
+          - distro-version: '38'
```

Comment:
I suggest opening a seperate pr with just this change -- there's likely a bit of work required to get this update passing which would best be done as a seperate effort.

Reply:
Normally, I already take care of some issues in #52638 regarding builing on fedora 38 and it should work out of the box. Still, I'll propose a separate PR. Unfortunately clang16 has landed in fedora 38 in the meantime and there is a version [incompatibility](https://bugzilla.redhat.com/show_bug.cgi?id=2187234) with clazy